pkg/cmd/pr/create/create.go

```
@@ -574,6 +577,23 @@ func NewCreateContext(opts *CreateOptions) (*CreateContext, error) {
 		}
 	}
 
+	if headRepo == nil && isPushEnabled && opts.PushToBaseRepo {
+		pushableRepos, err := repoContext.HeadRepos()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, r := range pushableRepos {
+			if ghrepo.IsSame(baseRepo, r) {
+				headRepo = r
+			}
+		}
+
+		if headRepo == nil {
+			return nil, fmt.Errorf("the user might lack write access to the base repository")
+		}
+	}
+
```

Should this be refactored into the following section to avoid duplicating logic but leveraging `(r *ResolvedRemotes) RemoteForRepo`, which is doing similar logic already?

https://github.com/cli/cli/blob/3443a752a9949ee4f954a84f7ab94a4b99df1b27/context/context.go#L161-L169

---

I'm probably missing something, but I don't see how we could incorporate the logic into the `(r *ResolvedRemotes) RemoteForRepo` method.

The proposed logic iterates over all pushable repos and determines if `baseRepo` can be pushed, just to fail early. Remotes, which `RemoteForRepo` handles, will only be used by the time we actually push on

https://github.com/cli/cli/blob/363dacbbed8cc9f86026ec1b1e38c6dca0d78c08/pkg/cmd/pr/create/create.go#L741-L743